### ISSUE TITLE
Set voldrv MDS slave state after a reconfiguration / failover

### ISSUE THREAD

**Issue by redlicha** [openvstorage/framework#567]
Scenario: volume uses MDS1 (master) and MDS2 (slave). A failover or on reconfigurations that swap the order MDS2 will be used by voldrv and promoted to master and MDS1 is left in a master role. This means it will not update itself from the backend anymore. In case of another swap of the configuration MDS1 will have to fetch too many TLogs from the backend. To make MDS1 update itself from the backend again, it n,eeds to be set into slave mode which must be protected against racing with a volumedriver failover by temporarily removing MDS1 from the config, setting it into slave mode and then adding it back to the config.

**Comment by wimpers** [comment]
@redlicha why can't the volumedriver tackle this?

**Comment by redlicha** [comment]
@wimpers: The main problem to solve is the failover case - in such a scenario volumedriver cannot notify / demote the previous master anymore obviously, so it would have to keep track of this "TODO-item" and try at a later point, once the instance can be reached again. In addition to that, the framework can also reconfigure the list of MDSes of a volume / force a failover to move the metadata closer to the volume.
IOW volumedriver would need to grow an MDS management component, while AFAIU the framework already has one that manages the placement of MDSes for example.

Creating such a mgmt component (that has a cluster-wide view on all MDSes) in voldrv might actually be worth investigating, but unless I'm overlooking something and/or grossly underestimating the effort (@khenderick or @kvanhijf: am I?), it should be easier to solve the issue for now in the framework.

**Comment by khenderick** [comment]
Issue find while validating changes for this ticket: #950.

**Comment by khenderick** [comment]
Implemented by #949, packaged in openvstorage-2.7.3-rev.4020.bdebc7c

**Comment by JeffreyDevloo** [comment]
### Information

Relying on unit testing to catch this for future releases.
Manually verifying this fix would prove to be difficult and consume a lot of time.

Unit test: test_role_assignment in /opt/OpenvStorage/ovs/lib/tests/mdsservice_tests/test_mdsservice

### Example output

```
###########################################################################################
# Processing test-module /opt/OpenvStorage/ovs/lib/tests/mdsservice_tests/test_mdsservice #
###########################################################################################

test_ensure_safety_of_2 (test_mdsservice.MDSServices) ... ok
test_ensure_safety_of_3 (test_mdsservice.MDSServices) ... ok
test_load_calculation (test_mdsservice.MDSServices) ... ok
test_mds_checkup (test_mdsservice.MDSServices) ... ok
test_role_assignments (test_mdsservice.MDSServices) ... ok
test_storagedriver_config_set (test_mdsservice.MDSServices) ... ok
test_syncreality (test_mdsservice.MDSServices) ... ok

----------------------------------------------------------------------
Ran 7 tests in 11.667s

OK



############
# OVERVIEW #
############

  - TestModule: /opt/OpenvStorage/ovs/lib/tests/mdsservice_tests/test_mdsservice  (7 tests)
    - DURATION: 11 seconds
    - SUCCESS: 7

```
